mrc-5970 lint, codecov and other friends

M-Kusumgar opened the pull request:
main changes:
* the npm package `codecov` is actually deprecated which explains a lot of the flakiness and problems, the recommendation is to use the github action which is what i have switch to here
* i have added a coverage command configured with istanbul provider
* set vitest to silent, vuex and vue warnings are very annoying in the buildkite stream
* removed check formatting for now, this is for a later time when we have a discussion about eslint rules and prettier formatting and what we want to move forward with, so i have also not included a lot of eslint rules from our original wodin file either, i think the defaults right now are actually stricter anyway

EmmaLRussell reviewed (commented):
I don't know if you're planning on reinstating the eslint rules, but rather than removing some valid exceptions here, I think it might be better to just retain some rules which it seems are now switched off. 

Specifically it feels like it would be good to retain the following to keep us tidy:
- no-explicit-any
- no-unused-vars
- no-setup-props-destructure

Not especially bothered about this others personally!

EmmaLRussell commented:
Coverage changes look great, good to deflakify that! :laughing:

M-Kusumgar commented:
check out the two latest commits for type changes!

M-Kusumgar commented:
have switched on the rules, i was planning to discuss the rules in more detail with you and anmol and prettier in this ticket https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-5973/eslint-and-prettier-discussion-ticket but have added those three rules as they seem uncontroversial (although no-setup-props-destructure is something we do a lot and i may lean towards not having that rule but happy to put it in right now)

i also went on a mission to rid the codebase of all the `any`s, felt great but also discovered a bug! we were using `object.details` instead of `object.detail` in `isApiError` so also fixed some tests related to it and fixed the code

EmmaLRussell reviewed (approved):
Great, thanks!
